internal/services/authorization/resource_user.go

```diff
@@ -159,7 +159,17 @@ func (r *UserResource) Create(ctx context.Context, req resource.CreateRequest, r
 
 	var plan *UserResourceModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	if resp.Diagnostics.HasError() {
+	if resp.Diagnostics.HasError() || plan == nil {
+		return
+	}
+
+	// Validate required fields before dereferencing
+	if plan.EnvironmentId.IsUnknown() || plan.EnvironmentId.IsNull() {
```

Comment:
[nitpick] The validation logic for EnvironmentId and AadId is repeated across Create, Read, Update, and Delete methods. Consider extracting these checks into a shared helper to reduce duplication and improve maintainability.